### ISSUE TITLE
1.37 breaks the fsprovider-sample extension

### ISSUE THREAD

**Issue by gjsjohnmurray** [microsoft/vscode#78961]
Issue Type: <b>Bug</b>

1. Install a fresh 1.36.1
2. Follow the README instructions at https://github.com/microsoft/vscode-extension-samples to clone the repo, then launch VS Code on the fsprovider-sample folder, then `npm install` and then `F5` to run the sample.
3. Follow the README at https://github.com/Microsoft/vscode-extension-samples/tree/master/fsprovider-sample to run the `MemFS: Setup Workspace` command in the Extension Host window.
4. Continue with those instructions and run `MemFS: Create Files`
5. Confirm that the explorer tree is populated.
6. Close the EH window without saving the workspace.
7. Upgrade the remaining VS Code window to 1.37.
8. Launch fsprovider-sample again using `F5`.
9. Run `MemFS: Setup Workspace`
10. Run `MemFS: Create Files` and observe the `command 'memfs.init' not found` error message.

The fact that an upgrade from 1.36.1 to 1.37.0 breaks this sample extension suggests the existence of a breaking change.

VS Code version: Code 1.37.0 (036a6b1d3ac84e5ca96a17a44e63a87971f8fcc8, 2019-08-08T02:33:50.993Z)
OS version: Windows_NT x64 10.0.17134

**Comment by gjsjohnmurray** [comment]
Another clue.

After step 9 save the workspace (File\Save Workspace As...). Observe the `Cannot read property 'replace' of undefined` message. Close the workspace, then reopen it from the .code-workspace file saving created. Now you can run the `MemFS: Create Files` command successfully.

**Comment by jrieken** [comment]
I have just tried with 1.37.0 and everything works as expected. 

> Upgrade the remaining VS Code window to 1.37.

⬆️ not sure what that means. Did you have different windows of the same app running different versions?

**Comment by jrieken** [comment]
@bpasero you mentioned an issue about missing activation events when restarting the extension host. does that apply here?

**Comment by jrieken** [comment]
I actually can reproduce...

**Comment by bpasero** [comment]
@jrieken there were 2 changes. The first is to do an atomic restart of the extension host (https://github.com/microsoft/vscode/commit/a383dc06076222cf7b40008703276ecd9677a74c) and the second a bugfix that happened when entering a workspace in an empty window running extension development (https://github.com/microsoft/vscode/commit/4968e0a712a564e2fd833ce4522c49e8b996fdb4). I tested the mem-fs extension specifically after fixing this, but only running out of sources. 

PS: my changes are only in insiders

**Comment by jrieken** [comment]
Yes, I cannot reproduce with 1.38. Closing as duplicate of such changes. @bpasero  leaving it up to you to decide if this should be fixed for 1.37.x

**Comment by gjsjohnmurray** [comment]
The error message in https://github.com/microsoft/vscode/issues/78809#issuecomment-520485794 looks similar to what I logged here, so perhaps the issues @bpasero has fixed in Insiders are having a wider effect on 1.37 users?

**Comment by gjsjohnmurray** [comment]
Confirming that it works for me too with today's 1.38.0-insider

**Comment by gjsjohnmurray** [comment]
For the record, duplicate of #77419. I vote for the fix to go into 1.37.1 Recovery.

**Comment by bpasero** [comment]
Opened https://github.com/microsoft/vscode/pull/79006
